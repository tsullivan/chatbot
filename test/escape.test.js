const request = require('supertest');
const { getServer } = require('../server');
const { utilFactory } = require('./utils');

const app = getServer();
const agent = request.agent(app);
const { handshake, send, checkResponses } = utilFactory(agent);

describe('escape-jail', () => {
  test('escape', async () => {
    await handshake(app);
    const resps = [];
    resps[resps.length] = await send('play escape_jail');
    resps[resps.length] = await send('items');
    resps[resps.length] = await send('escape');
    resps[resps.length] = await send('take_jail_key');
    resps[resps.length] = await send('force_key');
    resps[resps.length] = await send('wait');
    resps[resps.length] = await send('escape');
    resps[resps.length] = await send('escape');
    resps[resps.length] = await send('escape');
    resps[resps.length] = await send('jump');
    checkResponses(resps);
  });
});
