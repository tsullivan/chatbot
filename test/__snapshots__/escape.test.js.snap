// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`escape-jail escape 1`] = `
"# CRASH!!!

Your body has tumbled down a mountainside, and you have crashed through the roof of a jail. Fortunately you are not hurt, but you are stuck in this cell! The guards don't know that you are not supposed to be here, and they are watching you.

You are at: **The Jail Cell**

You're locked up securely in a jail cell. A hole in the ceiling marks where you crashed through accidentally when you got here.

There is a jail key on a far wall across the hallway from your jail cell.

- ESCAPE - Escape from the cell
- TAKE_JAIL_KEY - Somehow take the jail key from the wall.
- WAIT - Wait for the guards to stop watching you.
- LOOK - Look at the area you are currently in.
- ITEMS - Things you can do with stuff you are holding or stuff you see.
- HELP - See all the commands you can type
- QUIT - Quit the game"
`;

exports[`escape-jail escape 2`] = `"There aren't any items around or that you are holding that you can do anything with right now."`;

exports[`escape-jail escape 3`] = `
"You can not escape while the guards are watching!
---
You can:
- ESCAPE - Escape from the cell
- TAKE_JAIL_KEY - Somehow take the jail key from the wall.
- WAIT - Wait for the guards to stop watching you."
`;

exports[`escape-jail escape 4`] = `
"The key is on a wall far out of reach of your cell bars. You have to be really sneaky to take the it. Hmm. how to do this?

You hear a voice calling to you. \\"Use the FORCE, Tim.\\"

Who said that? The Force? What is that??
---
You can:
- USE_THE_FORCE_TO_TAKE_JAIL_KEY - Use the Force to take the jail cell key.
- - ESCAPE - Escape from the cell
- WAIT - Wait for the guards to stop watching you."
`;

exports[`escape-jail escape 5`] = `
"No time for questions. Time for escaping. You decide to put your trust in the mysterious voice. You close your eyes, and imagine the jail key floating up, off its hook, drift quietly through the air, and land in your open hand.

When you open your eyes, the jail key is in your hand.
---
You can:
- ESCAPE - Escape from the cell
- WAIT - Wait for the guards to stop watching you."
`;

exports[`escape-jail escape 6`] = `
"You pretend not to notice the guards watching you. They watch you for a little bit, then walk away to watch other prisoners.
---
You can:
- ESCAPE - Escape from the cell
- STAND_AROUND - Stand around and let the weight of feeling of excitement, exersion, and curiousity resolve itself in your mind before deciding on something clever to do"
`;

exports[`escape-jail escape 7`] = `
"You use the key to open the cell door, and quietly walk out.

You are at: **The Jail Hallway**

You're outside of the jail cell in a hallway lined with many other jail cells. The other jail cells have people that belong here, but you only got here accidentally. Time to get out!

There's a door at the end of the hallway - that must be the exit.

Just before the door, there is a window to the outside. You can see that you are on the ground floor of the jail, and there is a van right outside. There is a guard walking towards the driver's door.
---
You can:
- ESCAPE - Escape down the hallway and quickly jump into the back of the van
- STAND_AROUND - Stand around and let the weight of feeling of excitement, exersion, and curiousity resolve itself in your mind before deciding on something clever to do"
`;

exports[`escape-jail escape 8`] = `
"You sneak past the cells towards the door. Fortunately, no one sees you.

Thinking quickly, you quietly get outside, and get into the back of the van before the guard sees you.

You are at: **The Jail Van**

You're hiding in a jail van that's driving around the jail, making jail deliveries to the jail people.

You see:

- Jail Towel - A towel from the jail laundry van.
---
You can:
- TAKE_TOWEL - Take one of the towels from the jail laundry van.
- - ESCAPE - Escape the jail van
- STAND_AROUND - Stand around and let the weight of feeling of excitement, exersion, and curiousity resolve itself in your mind before deciding on something clever to do"
`;

exports[`escape-jail escape 9`] = `
"You sneak out of the jail van and run over to a fence. You're almost out of here!

You are at: **The Jail Fence**

You've just escaped from a jail. You breathe a sigh of relief because that part of your life is forever behind you. They'll never find you in these woods. Plus, you stole the jail key, because you're not a bad guy. You were put in jail accidentally. They made you steal the key.

Now, all you have to do is get over this fence...

To be honest, it's a little scary right now because you really don't know what's out here. And there's cops chasing you, but they're far behind. You have the Force and you can go super fast...

You're at the huge fence that goes all around the jail and keeps the prisoners inside. It's very tall and you would have to be a really good climber or a really good jumper, or both, to get over this thing. And you would have to be really really good to do it without getting noticed by a guard!
---
You can:
- JUMP_THE_FENCE - Jump over the jail fence
- CLIMB_THE_FENCE - Climb over the jail fence"
`;

exports[`escape-jail escape 10`] = `
"JUUUUMP! Oh no! The top of the fence is still too high up to jump over.
---
You can:
- CLIMB_THE_FENCE - Climb over the jail fence"
`;
